backend/remote/backend.go

```diff
@@ -211,21 +212,30 @@ func (b *Remote) Configure(obj cty.Value) tfdiags.Diagnostics {
 	}
 
 	// Discover the service URL for this host to confirm that it provides
-	// a remote backend API and to discover the required base path.
-	service, err := b.discover(b.hostname)
+	// a remote backend API and to get the version constraints.
+	service, constraints, err := b.discover()
 	if err != nil {
 		diags = diags.Append(tfdiags.AttributeValue(
 			tfdiags.Error,
 			strings.ToUpper(err.Error()[:1])+err.Error()[1:],
 			"", // no description is needed here, the error is clear
 			cty.Path{cty.GetAttrStep{Name: "hostname"}},
 		))
+	}
+
+	// Check any retrieved constraints to make sure we are compatible.
+	if constraints == nil {
```

Comment:
I think this should be `!=`, no?

Reply:
Nice catch! I added this later just to prevent calling the `checkConstraints` function when it’s not needed (in ‘checkContraints` it also checks if the passed in value is `nil`)

But now it will only be called when it shouldn’t, so thanks! Will update it in a bit and add a test covering this as well 👍🏻

Reply:
@blalor PR #19667 adds a fix and test for this